R/migrate.project.R

```diff
@@ -17,118 +17,127 @@
-          if(grepl("cache_loaded_data", config_warnings)) {
-                  # switch the setting to FALSE so as to not mess up any existing
-                  # munge script, but warn the user
-                  loaded.config$cache_loaded_data <- FALSE
-                  message(paste0(c(
-                          "\n",
-                          "There is a new config item called cache_loaded_data which auto-caches data",
-                          "after it has been loaded from the data directory.  This has been switched",
-                          "off for this project in case it breaks your scripts.  However you can switch",
-                          "it on manually by editing global.dcf"),
-                          collapse="\n"))
-          }
-          
+    # Tell the user about problems with their old config
+    message(paste0(c(
+      "Your existing project configuration in globals.dcf does not contain up to",
+      paste0("date configuration settings in this version ",
+             .package.version(),
+             " of ProjectTemplate.  They will"),
+      "be added automatically during migration, but you should review afterward."
+    ),
+    collapse = "\n"))
+
+    if (grepl("missing a configuration file", config_warnings)) {
+      message(paste0(c(
+        "You didn't have a config.dcf file.  One has been created",
+        "for you using default values"
+      ),
+      collapse = "\n"))
+    }
+
+    if (grepl("missing the following entries", config_warnings)) {
+      message(paste0(c(
+        "Your config.dcf file was missing entries and defaults",
+        "have been used.  The missing entries are:"
+      ),
+      collapse = "\n"))
+      missing <- sub(".*missing the following entries:([^.]*)\\.(.*)$", "\\1", config_warnings)
+      message(missing)
+    }
+
+    if (grepl("contains the following unused entries", config_warnings)) {
+      message(paste0(c(
+        "Your config.dcf file contained unused entries which have been",
+        "removed.  The unused entries are:"
+      ),
+      collapse = "\n"))
+      unused <- sub(".*contains the following unused entries:([^.]*)\\.(.*)$", "\\1", config_warnings)
+      message(unused)
+    }
+
+    # Specific logic here for new config items that need special migration treatment
+    if (grepl("cache_loaded_data", config_warnings)) {
+      # switch the setting to FALSE so as to not mess up any existing
+      # munge script, but warn the user
+      loaded.config$cache_loaded_data <- FALSE
+      message(paste0(c(
+        "\n",
+        "There is a new config item called cache_loaded_data which auto-caches data",
+        "after it has been loaded from the data directory.  This has been switched",
+        "off for this project in case it breaks your scripts.  However you can switch",
+        "it on manually by editing global.dcf"),
+        collapse = "\n"))
+    }
   }
-  
+
   # Process other migration conflicts
-  
-  
-  
+  if (doc_not_renamed) {
```

Comment:
I'm confused here - shouldn't it be `directory_conflict`?

In fact, aren't these two always going to be the same?  One looks redundant.